js/logic_dictionary.js

```diff
@@ -26,11 +26,21 @@ utilities.dictionary = (function () {
     return encountered[symbol];
   }
 
+  function getLearnedWord(encounter) {
+    return encounter.actual;
+  }
+
   function learnRandomWord() {
+    var learnedWords = _.chain(encountered)
+      .map(getLearnedWord)
+      .compact().value();
     var getSymbol = utilities.symbols.getSymbol;
     var getRandomWords = utilities.symbols.getRandomWords;
     var getRandomItemWords = utilities.symbols.getRandomItemWords;
-    var words = _.concat(getRandomWords(2), getRandomItemWords(2));
+    var randomWords = _.concat(getRandomWords(2), getRandomItemWords(2));
```

Comment:
Hmm ... my reading of this says that because there is a sampling that happens in getRandomWords, it will be possible for us to not get a word even if there are still unknown words?

Reply:
It's sort of like this: a human knows a sampling of 2-4 words. If you already know them, then you won't learn a new word.

Reply:
I'm just wondering if this might get a bit tedious from the gameplay perspective. Like, I suspect it will make it really hard to learn the full vocabulary? My proposal is to have getRandomWords and getRandomItem words do the difference before they call sampleSize.

Reply:
My only concern is that it's fairly easy to do a strategy where you just continue to probe until you get the full dictionary. Our game doesn't have a turn limit. I think having some diminishing returns on probing is good for gameplay.

Reply:
Ah I see. Ok. That's ... probably reasonable. We may need to tweak the constants, but I get it. =P